lib/pos_fee_estimator.go

```diff
@@ -0,0 +1,159 @@
+package lib
+
+import (
+	"github.com/deso-protocol/core/collections"
+	"github.com/pkg/errors"
+	"sort"
+)
+
+type PoSFeeEstimator struct {
+	mempoolFeeEstimatorMempool    *PosMempool
+	pastBlocksTransactionRegister *TransactionRegister
+	numBlocks                     uint64
+	cachedBlocks                  []*MsgDeSoBlock
+}
+
+func (posFeeEstimator *PoSFeeEstimator) Init(mempool *PosMempool, pastBlocks []*MsgDeSoBlock, numBlocks uint64) error {
+	if mempool == nil {
```

Comment:
Worth adding a lock here too
```suggestion
	posFeeEstimator.lock.Lock()
	defer posFeeEstimator.lock.Unlock()

	if mempool == nil {
```